src/app/charts/line-graph.component.ts

```diff
@@ -132,6 +131,12 @@ export class LineGraphComponent {
         // Time scales only recognize annual and daily data
         var parseTime = D3.timeParse(this.timeFormat);
         this.extractedData.forEach(d => d.date = parseTime(d.date));
+
+        // Sort data by date ascending
+        this.extractedData.sort(function(a, b) {return a.date - b.date;});
```

Comment:
I see your point. There isn't a reason to maintain the string format of `date` that comes from the API. It makes sense to process date in the service while we are chopping up the data.

I could define two types: RawDataPoint and DataPoint. Former, string date; Latter, date date.